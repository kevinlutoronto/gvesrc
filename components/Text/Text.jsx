// @flow
import React from 'react';
import cx from 'classnames';
import style from './style.scss';
import type { Data, StyleAttributes } from '../../types';
import { getDataAttrs } from '../../dataUtils';

export type Props = {|
  children: React$Node,
  data?: Data,
  large?: boolean,
  style?: StyleAttributes,
  type?: string,
|};

const types = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
};

class Text extends React.Component<Props> {
  static defaultProps = {
    style: { width: '200px', height: 'auto' },
  };

  render() {
    const { large, type } = this.props;
    const textStyle = cx(style.wrapper, {
      [style.large]: large,
      [style.primary]: type === types.PRIMARY,
      [style.secondary]: type === types.SECONDARY,
    });
    return (
      <div {...getDataAttrs(this.props.data)} className={textStyle} style={this.props.style}>
        {this.props.children}
      </div>
    );
  }
}

Text.displayName = 'Plasma@Text';

export default Text;
