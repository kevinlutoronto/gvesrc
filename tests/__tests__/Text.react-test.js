/* eslint-disable */

import React from 'react';
import Text from '../../components/Text/Text';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Text Component', () => {
  test('render', () => {
    const component = renderer.create(<Text />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
