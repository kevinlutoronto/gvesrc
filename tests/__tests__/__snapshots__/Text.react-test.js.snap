// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component render 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": "auto",
      "width": "200px",
    }
  }
/>
`;
